src/scss/parse.js

```diff
@@ -4038,106 +4023,257 @@ function _checkUnicodeWildcard(i) {
 }
 
 /**
- * Check if token is part of URI (e.g. `url('/css/styles.css')`)
- * @param {Number} i Token's index number
- * @returns {Number} Length of URI
+ * Check if token is part of URI, e.g. `url('/css/styles.css')`
+ * @param {number} i Token's index number
+ * @returns {number} Length of URI
  */
 function checkUri(i) {
-  var start = i;
+  const start = i;
 
-  if (i >= tokensLength || tokens[i++].value !== 'url' ||
-      i >= tokensLength || tokens[i].type !== TokenType.LeftParenthesis)
-      return 0;
+  if (i >= tokensLength || tokens[i].value !== 'url') return 0;
 
-  return tokens[i].right - start + 1;
+  // Skip url
+  i++;
+
+  if (i >= tokensLength || tokens[i].type !== TokenType.LeftParenthesis)
+    return 0;
+
+  // Store the opening parenthesis token as we will reference it's `right`
+  // property to determine when the parentheses close
+  const leftParenthesis = tokens[i];
+
+  // Skip (
+  i++;
+
+  // Determine the type of URI
+  if (checkUri1(i)) tokens[start].uriType = 1; // Raw based URI
+  else if (checkUri2(i)) tokens[start].uriType = 2; // Non-raw based URI
+  else return 0;
```

Comment:
Current implementation doesn't guarantee that those `checkUri` checks will go through all characters between parentheses because we don't compare token position with a matching right parenthesis anywhere. In `checkUri1` for example we check for `i < tokensLength` in while loop. I suppose that will parse the following example as valid even though it is not: `url(abc(()`